WordPress/Classes/ViewRelated/Stats/Period Stats/SiteStatsPeriodTableViewController.swift

```diff
@@ -193,14 +173,12 @@ private extension SiteStatsPeriodTableViewController {
     }
 
     func refreshData() {
-        guard let selectedDate = selectedDate,
-            let selectedPeriod = selectedPeriod,
-            viewIsVisible() else {
+        guard viewIsVisible() else {
             refreshControl.endRefreshing()
-                return
+            return
         }
         addViewModelListeners()
-        viewModel?.refreshTrafficOverviewData(withDate: selectedDate, forPeriod: selectedPeriod)
+        viewModel.refreshTrafficOverviewData(withDate: datePickerViewModel.date, forPeriod: datePickerViewModel.period)
```

Comment:
According to requirements (https://github.com/wordpress-mobile/jetpack-issue-repo/issues/1), we need to:

> Remember the last selected tab and date range (as we do currently with Day/Week/Month/Year tabs)

It's not working with this PR but I have implemented it in https://github.com/wordpress-mobile/WordPress-iOS/pull/22695 and the same solution should continue working.

Reply:
Thanks for catching and fixing that!